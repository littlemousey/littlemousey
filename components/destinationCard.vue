<template>
  <v-col cols="12" sm="5" md="3">
    <v-card hover nuxt :to="link">
      <v-img :src="imageUrl"></v-img>

      <v-card-title primary-title>
        <div>
          {{ title }}
        </div>
        <div class="grey--text subtitle-1">{{ text }}</div>
      </v-card-title>

      <v-card-actions>
        <v-btn text color="orange" small nuxt :to="link">{{
          buttonText
        }}</v-btn>
      </v-card-actions>
    </v-card>
  </v-col>
</template>

<script>
export default {
  name: 'DestinationCard',
  props: {
    title: {
      type: String,
      required: true
    },
    text: {
      type: String,
      required: true
    },
    imageUrl: {
      type: String,
      required: true
    },
    link: {
      type: String,
      default: ''
    },
    buttonText: {
      type: String,
      default: ''
    }
  }
}
</script>
