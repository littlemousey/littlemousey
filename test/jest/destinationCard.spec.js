import { shallowMount } from '@vue/test-utils'
import Vue from 'vue'
import Vuetify from 'vuetify'
import destinationCard from '@/components/destinationCard'

Vue.use(Vuetify)

let wrapper

beforeEach(() => {
  wrapper = shallowMount(destinationCard, {
    propsData: {
      title: 'title',
      text: 'text',
      imageUrl: 'imageUrl'
    }
  })
})

afterEach(() => {
  wrapper.destroy()
})

describe('Destination card', () => {
  it('renders correctly', () => {
    expect(wrapper.element).toMatchSnapshot()
  })
})
