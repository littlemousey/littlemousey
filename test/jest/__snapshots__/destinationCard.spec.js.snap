// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`Destination card renders correctly 1`] = `
<v-col-stub
  cols="12"
  md="3"
  sm="5"
  tag="div"
>
  <v-card-stub
    hover="true"
    loaderheight="4"
    nuxt="true"
    tag="div"
    to=""
  >
    <v-img-stub
      position="center center"
      src="imageUrl"
      transition="fade-transition"
    />
     
    <v-card-title-stub
      primary-title=""
      primarytitle=""
    >
      <div>
        
        title
      
      </div>
       
      <div
        class="black--text subtitle-1"
      >
        text
      </div>
    </v-card-title-stub>
     
    <v-card-actions-stub>
      <v-btn-stub
        activeclass=""
        color="orange"
        nuxt="true"
        small="true"
        tag="button"
        text="true"
        to=""
        type="button"
      >
        
      </v-btn-stub>
    </v-card-actions-stub>
  </v-card-stub>
</v-col-stub>
`;
